clang/lib/AST/CXXInheritance.cpp

```diff
@@ -265,8 +268,8 @@ bool CXXBasePaths::lookupInBases(ASTContext &Context,
             BaseRecord = nullptr;
         }
       } else {
-        BaseRecord = cast<CXXRecordDecl>(
-            BaseSpec.getType()->castAs<RecordType>()->getDecl());
+        if (auto *RT = BaseSpec.getType()->getAs<RecordType>())
+          BaseRecord = cast<CXXRecordDecl>(RT->getDecl());
```

Comment:
This should be equivalent to:
```suggestion
        BaseRecord = BaseSpec.getType()->getAsRecordDecl();
```

Reply:
This is almost same. To be correct, we need to replace the first variant with 
```c++
BaseRecord = cast<CXXRecordDecl>(BaseSpec.getType()->getAsRecordDecl());
```

All in all, thanks for this comment. Fixed